File: src/Bionic.Bridge.Capacitor.JS/src/ModalsBridge.ts
import {
  ActionSheetOptions,
  ActionSheetResult,
  AlertOptions,
  ConfirmOptions,
  ConfirmResult,
  Plugins,
  PromptOptions,
  PromptResult
} from '@capacitor/core';

const {Modals} = Plugins;

interface IModals {
  alert(options: AlertOptions): Promise<void>;

  confirm(options: ConfirmOptions): Promise<ConfirmResult>;

  prompt(options: PromptOptions): Promise<PromptResult>;

  showActions(options: ActionSheetOptions): Promise<ActionSheetResult>;
}

export class ModalsBridge implements IModals {

  public async alert(options: AlertOptions): Promise<void> {
    if (!Modals) return Promise.reject("Modals not available.");
    try {
      return await Modals.alert(options);
    } catch (e) {
      return new Promise<void>((r, reject) => reject(e.message));
    }
  }

  public async confirm(options: ConfirmOptions): Promise<ConfirmResult> {
    if (!Modals) return Promise.reject("Modals not available.");
    try {
      return await Modals.confirm(options);
    } catch (e) {
      return new Promise<ConfirmResult>((r, reject) => reject(e.message));
    }
  }

  public async prompt(options: PromptOptions): Promise<PromptResult> {
    if (!Modals) return Promise.reject("Modals not available.");
    try {
      return await Modals.prompt(options);
    } catch (e) {
      return new Promise<PromptResult>((r, reject) => reject(e.message));
    }
  }

  public async showActions(options: ActionSheetOptions): Promise<ActionSheetResult> {
    if (!Modals) return Promise.reject("Modals not available.");
    try {
      return await Modals.showActions(options);
    } catch (e) {
      return new Promise<ActionSheetResult>((r, reject) => reject(e.message));
    }
  }
}

File: src/Bionic.Bridge.Capacitor/ModalsBridge.cs
using System.Threading.Tasks;
using Microsoft.JSInterop;

namespace Bionic.Bridge.Capacitor {
    public static class ModalsBridge {
        private const string ModalsAlertFunctionName = "BionicBridge.Capacitor.Modals.alert";
        private const string ModalsConfirmFunctionName = "BionicBridge.Capacitor.Modals.confirm";
        private const string ModalsPromptFunctionName = "BionicBridge.Capacitor.Modals.prompt";
        private const string ModalsShowActionsFunctionName = "BionicBridge.Capacitor.Modals.showActions";

        public static Task Alert(AlertOptions options) =>
            JSRuntime.Current.InvokeAsync<object>(ModalsAlertFunctionName, options);

        public static Task<ConfirmResult> Confirm(ConfirmOptions options) =>
            JSRuntime.Current.InvokeAsync<ConfirmResult>(ModalsConfirmFunctionName, options);

        public static Task<PromptResult> Prompt(PromptOptions options) =>
            JSRuntime.Current.InvokeAsync<PromptResult>(ModalsPromptFunctionName, options);

        public static Task<ActionSheetResult> ShowActions(ActionSheetOptions options) =>
            JSRuntime.Current.InvokeAsync<ActionSheetResult>(ModalsShowActionsFunctionName, options);
    }

    public class AlertOptions {
        public string title;
        public string message;
        public string buttonTitle;
    }

    public class ConfirmOptions {
        public string title;
        public string message;
        public string okButtonTitle;
        public string cancelButtonTitle;
    }

    public class ConfirmResult {
        public bool value;
    }

    public class PromptOptions {
        public string title;
        public string message;
        public string okButtonTitle;
        public string cancelButtonTitle;
        public string inputPlaceholder;
    }

    public class PromptResult {
        public bool cancelled;
        public string value;
    }

    public class ActionSheetOptions {
        public string title;
        public string message;
        public ActionSheetOption[] options;
    }

    public class ActionSheetOption {
        public string title;
        public string style;
        public string icon;
    }

    public class ActionSheetResult {
        public long index;
    }

    public static class ActionSheetOptionStyle {
        public static string Default = "DEFAULT";
        public static string Destructive = "DESTRUCTIVE";
        public static string Cancel = "CANCEL";
    }
}